pkg/ipmi/ipmi_integration_test.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019-2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ipmi
+
+import (
+	"testing"
+
+	"github.com/u-root/u-root/pkg/qemu"
+	"github.com/u-root/u-root/pkg/vmtest"
+)
+
+func TestIntegration(t *testing.T) {
+	o := &vmtest.Options{
+		QEMUOpts: qemu.Options{
+			Devices: []qemu.Device{
+				// This integration test requires kernel built with the following options set:
+				// CONFIG_IPMI=y
+				// CONFIG_IPMI_DEVICE_INTERFACE=y
+				// CONFIG_IPMI_WATCHDOG=y
+				// CONFIG_IPMI_SI=y
+				qemu.ArbitraryArgs{"-device", "ipmi-bmc-sim,id=bmc0"},
+				qemu.ArbitraryArgs{"-device", "isa-ipmi-kcs,bmc=bmc0,irq=5"},
+			},
+		},
+	}
+	vmtest.GolangTest(t, []string{"github.com/u-root/u-root/pkg/ipmi"}, o)
```

Comment:
[Integration/Readme](https://github.com/u-root/u-root/blob/main/integration/README.md) and [Contributing](https://github.com/u-root/u-root/blob/main/CONTRIBUTING.md) as well as looking at pkg/mount didn't help me figuring out how to use it.

Reply:
Will look into other pkgs and hope to see a pattern how to use it.

Reply:
This may take longer, because I have to figure out how to handle the behaviour of the bmc simulator and the reactions of the vm.

Reply:
I'd start with all the tests that are in `ipmi_test.go` placed in `ipmi_linux_test.go`, replace `i := GetMockIPMI()` with `i, err := Open(0)` 

and see what happens when that runs in the VM. some of them will probably fail because the QEMU BMC doesn't support those features, but you can go from there.

(And an intentionally failing test of something QEMU doesn't support would be appreciated, actually, too, just to tickle some failure cases.)

Reply:
Thanks for the hint